lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -473,6 +478,43 @@ def cmd_exit(*args)
 
   alias cmd_quit cmd_exit
 
+
+  def cmd_history(*args)
+    
+    limit = @history_limit
+    length = Readline::HISTORY.length 
+
+    @@history_opts.parse(args) do |opt, _idx, val|
+      case opt
+      when "-a"
+        limit = length
+      when "-n"
+        return cmd_history_help unless val.match /\A[-+]?\d+\z/
```

Comment:
`history -n`
This throws a call stack trace for undefined method 'match' for nil:NilClass.  Check `val` before running match against it.